config/config_test.go

```diff
@@ -140,12 +141,17 @@ func TestConfig_LoadDirectory(t *testing.T) {
 	expectedConfigs[0].Tags = make(map[string]string)
 
 	expectedPlugins[1] = inputs.Inputs["exec"]().(*MockupInputPlugin)
-	p, err := parsers.NewParser(&parsers.Config{
+	parserConfig := &parsers.Config{
 		MetricName: "exec",
 		DataFormat: "json",
 		JSONStrict: true,
-	})
+	}
+	p, err := parsers.NewParser(parserConfig)
 	require.NoError(t, err)
+
+	// Inject logger to have proper struct for comparison
+	models.SetLoggerOnPlugin(p, models.NewLogger("parsers", parserConfig.DataFormat, parserConfig.MetricName))
```

Comment:
Why not using the `testutil.Logger` here?

Reply:
I think it is more complicated. Let me explain.

In this PR I introduced `Log` in `plugins/parsers/json/parser.go`. Thus `exec` input plugin created in this test here: `require.NoError(t, c.LoadDirectory("./testdata/subconfig"))` contains `json` parser which contains non-nil `Log` (it was automatically injected by Telegraf's core).

Without any change in this file, test would fail here: 
`require.Equalf(t, expectedPlugins[i], plugin.Input, "Plugin %d: incorrect struct produced", i)` 
because `expectedPlugins[1]` (`exec` input plugin) would contain `json` parser without injected `Log` (it would be `nil`).

To make `expectedPlugins[i]` and `plugin.Input` equal I need to either inject `Log` into `expectedPlugins[i]` (which I did) or set `Log` to `nil` in `plugin.Input` (which I did not, because I didn't find any clean solution for that).

Reply:
Yeah I face the same issue in the parser restructure and try to set the `Log` to `nil`. But this is by far a more elegant solution to the problem. Will steal this for the parser restructure... ;-P

Thanks for the explanation!